### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by baudlord**

### CODE REVIEW

File: manifests/manager.pp [hunk 1]
```diff
@@ -231,21 +222,16 @@
...
-    # TODO: Allow filtering of EPEL requirement
-    class { 'wazuh::repo':}
-    if $::osfamily == 'Debian' {
-      Class['wazuh::repo'] -> Class['apt::update'] -> Package[$wazuh::params_manager::server_package]
-    } else {
-      Class['wazuh::repo'] -> Package[$wazuh::params_manager::server_package]
-    }
-  }
+  ## Install wazuh-repository
+  #if $manage_repos {
```
Comment (Wiston999) [inline manifests/manager.pp]: Uncomment?

File: manifests/manager.pp [hunk 2]
```diff
@@ -335,7 +292,6 @@
           content => template($ossec_rootcheck_template);
       }
   }
-
   if ($configure_wodle_openscap == true){
```
Comment (Wiston999) [inline manifests/manager.pp]:
```suggestion
  if $configure_wodle_openscap {
```
Checking if a bool is `== true` is non-sense

File: templates/fragments/_sca.erb [hunk 3]
```diff
@@ -1,70 +1,39 @@
...
     <sca>
       <enabled>yes</enabled>
       <scan_on_start>yes</scan_on_start>
       <interval>12h</interval>
       <skip_nfs>yes</skip_nfs>
 
       <policies>
+      <%- if @os['family'] == 'CentOS' -%>
         <policy>cis_rhel7_linux_rcl.yml</policy>
+      <%- elsif @os['family'] == 'Amazon' -%>
```
Comment (Wiston999) [inline templates/fragments/_sca.erb]: empty if body, remove this line

File: templates/fragments/_sca.erb [hunk 4]
```diff
@@ -1,70 +1,39 @@
...
       <interval>12h</interval>
       <skip_nfs>yes</skip_nfs>
 
       <policies>
+      <%- if @os['family'] == 'CentOS' -%>
         <policy>cis_rhel7_linux_rcl.yml</policy>
+      <%- elsif @os['family'] == 'Amazon' -%> 
+      <%- elsif @os['family'] =='RedHat' -%> 
+        <%- if @os['release']['major'] == '7' -%>
+        <policy>cis_rhel7_linux_rcl.yml</policy>
```
Comment (Wiston999) [inline templates/fragments/_sca.erb]:
```suggestion
        <policy>cis_rhel<%= @os['release']['major'] %>_linux_rcl.yml</policy>
```
and remove all the if / elsif checking for major release version.

File: templates/fragments/_sca.erb [hunk 5]
```diff
@@ -1,70 +1,39 @@
...
-  <%- if @apply_template_os == 'centos' -%>
     <sca>
       <enabled>yes</enabled>
       <scan_on_start>yes</scan_on_start>
       <interval>12h</interval>
       <skip_nfs>yes</skip_nfs>
 
       <policies>
+      <%- if @os['family'] == 'CentOS' -%>
         <policy>cis_rhel7_linux_rcl.yml</policy>
```
Comment (Wiston999) [inline templates/fragments/_sca.erb]: Only centos 7 is supported?

File: templates/fragments/_sca.erb [hunk 6]
```diff
@@ -1,70 +1,39 @@
...
+        <%- if @os['release']['major'] == '7' -%>
+        <policy>cis_rhel7_linux_rcl.yml</policy>
+        <%- elsif @os['release']['major'] == '6' -%>
+        <policy>cis_rhel6_linux_rcl.yml</policy>
+        <%- elsif @os['release']['major'] == '5' -%>
+        <policy>cis_rhel5_linux_rcl.yml</policy>
+        <%- end -%>
+      <%- elsif @os['family'] == 'Debian' -%>
+        <%- if @os['release']['major'] == '7' -%>
+        <policy>cis_debian7_L1.yml</policy>
```
Comment (Wiston999) [inline templates/fragments/_sca.erb]: Same as RHEL:
```suggestion
        <policy>cis_debian<%= @os['release']['major'] %>_L1.yml</policy>
```
and remove checks for each version.

File: templates/fragments/_sca.erb [hunk 7]
```diff
@@ -1,70 +1,39 @@
...
-  <%- else -%>
-      <sca>
-      <enabled>yes</enabled>
-      <scan_on_start>yes</scan_on_start>
-      <interval>12h</interval>
-      <skip_nfs>yes</skip_nfs>
-
-      <policies>
-      <policy>cis_debian_linux_rcl.yml</policy>
-      <%- if @debian_additional_templates == 'yes' -%>
```
Comment (Wiston999) [inline templates/fragments/_sca.erb:58]: This has been removed, on purpose?

File: templates/fragments/_sca.erb [hunk 8]
```diff
@@ -1,70 +1,39 @@
...
-      <policy>cis_debianlinux7-8_L1_rcl.yml</policy>
```
Comment (Wiston999) [inline templates/fragments/_sca.erb:59]: New version uses different files for debian, is that correct?

File: templates/fragments/_wodle_vulnerability_detector.erb [hunk 9]
```diff
@@ -1,5 +1,5 @@
   <wodle name="vulnerability-detector">
-    <disabled>yes</disabled>
+    <disabled><%= @wodle_vulnerability_detector_ %></disabled>
     <interval>5m</interval>
```
Comment (Wiston999) [inline templates/fragments/_wodle_vulnerability_detector.erb]: Parametrize rest of vars?

**Review by Wiston999 — CHANGES_REQUESTED** [review]

File: templates/fragments/_sca.erb (hunk 5 above)
Comment (baudlord) [reply templates/fragments/_sca.erb]: This was WIP, uses the same ruleset as RHEL, it's fixed now.

File: templates/fragments/_wodle_vulnerability_detector.erb (hunk 9 above)
Comment (baudlord) [reply templates/fragments/_wodle_vulnerability_detector.erb]: WIP!

File: templates/fragments/_sca.erb (hunk 3 above)
Comment (baudlord) [reply templates/fragments/_sca.erb]: The empty body is used to disable EL rules.

Amazon Linux matches with family RedHat, but the EL rules are not used upstream, so I've disabled them here as well.

https://github.com/wazuh/wazuh-puppet/blob/master/templates/fragments/_sca.erb#L16

**Pull request closed without merging**